src/zodbupdate/main.py

```diff
@@ -192,6 +192,13 @@ def main():
     # doesn't complain.
     if args.convert_py3 and six.PY3 and not args.dry_run:
         zodbupdate.convert.update_magic_data_fs(args.file)
+    elif args.convert_py3 and six.PY3 and args.dry_run:
+        zodb_magic = zodbupdate.utils.get_zodb_magic(args.file)
+        if zodb_magic != ZODB.FileStorage.packed_version:
+            raise AssertionError(
+                'You cannot use --dry-run under Python 3 with a ZODB '
+                'created under Python 2 as --dry-run does not rewrite the '
+                'magic header data before opening the ZODB file.')
```

Comment:
I'm not sure AssertionError is the right one to raise.  How about SystemExit (i.e. `sys.exit("You cannot ...")`)?  There shouldn't be any tracebacks shown to the user, just the error message.

Reply:
Right, I didn't even think about that. I just used AssertionError because that was used elsewhere in the code to signal catastrophic failure during invocation. Changed it now.